chia/cmds/param_types.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from decimal import Decimal, InvalidOperation
+from typing import Any, Optional, Union
+
+import click
+
+from chia.cmds.units import units
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.bech32m import bech32_decode, decode_puzzle_hash
+from chia.util.config import selected_network_address_prefix
+from chia.util.ints import uint64
+from chia.wallet.util.address_type import AddressType
+
+
+class TransactionFeeParamType(click.ParamType):
+    """
+    A Click parameter type for transaction fees, which can be specified in XCH or mojos.
+    """
+
+    name: str = "uint64"  # output type
+    value_limit: Decimal = Decimal(0.5)
+
+    def convert(self, value: Any, param: Optional[click.Parameter], ctx: Optional[click.Context]) -> uint64:
+        if isinstance(value, uint64):  # required by click
+            return value
```

Comment:
the click docs say you always have to do the isinstance check.

Reply:
it makes me a bit uneasy, but I suppose that's how it's meant to be used. Please double-check that you have test coverage of all cases where one may mix up the units for this

Reply:
I see the doc comments about this being the thing to do.  https://click.palletsprojects.com/en/8.1.x/api/#click.ParamType.convert  I'm not sure it is actually mandatory vs. being a practice to help deal with letting folks write defaults like `default=123` instead of `default="123"`.  I suspect that there are other corner cases where it matters as well.  Maybe when passing something like `123` to `CliRunner.invoke()` instead of `"123"`?

I agree with there being some risk here which makes me curious to know if it works, in our specific uses, without this type check.  Or perhaps replaced with a requirement that the input be a string.  Then we'll get notified if we hit some corner case and we can then consider how to handle it instead of it 'just working' possibly in some unexpected way.

It appears that perhaps since this comment there has been a change below to put off implementing support for mojos, maybe.  I think the general question of how we should handle this remains.

Reply:
Should we try just failing on it not being a string instance and see what we learn?

Reply:
This looks unchanged.  It's been awhile since I had all the nuances here in my head.  Do we want to try to have a call about this with the three of us to refresh etc?

Reply:
removed when appropriate as discussed